base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -952,20 +965,27 @@ where
         if completed_tx.status != TransactionStatus::Completed || completed_tx.transaction.body.kernels().is_empty() {
             return Err(TransactionServiceError::InvalidCompletedTransaction);
         }
+        let timeout = match self.power_mode {
+            PowerMode::LowPower => self.config.base_node_monitoring_timeout,
+            PowerMode::NormalPower => self.config.low_power_polling_timeout,
+        };
         match self.base_node_public_key.clone() {
             None => return Err(TransactionServiceError::NoBaseNodeKeysProvided),
             Some(pk) => {
-                let (mempool_response_sender, mempool_response_receiver) = mpsc::channel(100);
-                let (base_node_response_sender, base_node_response_receiver) = mpsc::channel(100);
+                let (mempool_response_sender, mempool_response_receiver) = mpsc::channel(500);
+                let (base_node_response_sender, base_node_response_receiver) = mpsc::channel(500);
+                let (timeout_update_sender, timeout_update_receiver) = mpsc::channel(100);
                 self.mempool_response_senders.insert(tx_id, mempool_response_sender);
                 self.base_node_response_senders.insert(tx_id, base_node_response_sender);
+                self.timeout_update_senders.insert(tx_id, timeout_update_sender);
```

Comment:
Wouldn't a single timeout event for the txn system work and be more efficient, rather than per transaction? Would, of course, need a broadcast channel to produce multiple receivers. 

One important note, with a broadcast channel the underlying Vec is preallocated to the given buffer size. 100 seems too large for how often the power mode will be changed - 5 or 10 should be plenty (`std::mem::size_of::<Duration>() == 16`(1600 bytes vs 160 bytes)

Reply:
Ja, I agree its neater. Was just in this mindset for the other channels.